### ISSUE TITLE
building error 

### ISSUE THREAD

**Issue by GitPaean** [OPM/dune-cornerpoint#201]
Guessing related to OPM/opm-parser#661. There are probably some other compiling errors following the following one. 

```
153 /home/kaib/OPM/debug/dune-cornerpoint/dune/grid/common/ZoltanGraphFunctions.cpp: In constructor ‘Dune::cpgrid::CombinedGridWellGraph::CombinedGridWellGraph(const Dune::CpGrid&, Opm::EclipseStateConstPtr, con    st double*, bool)’:
154 /home/kaib/OPM/debug/dune-cornerpoint/dune/grid/common/ZoltanGraphFunctions.cpp:343:72: error: invalid use of incomplete type ‘const class Opm::Schedule’
155      std::vector<Opm::WellConstPtr> wells  = eclipseState->getSchedule()->getWells();
156                                                                         ^
157 In file included from /home/kaib/OPM/debug/dune-cornerpoint/dune/grid/common/ZoltanGraphFunctions.hpp:24:0,
158                  from /home/kaib/OPM/debug/dune-cornerpoint/dune/grid/common/ZoltanGraphFunctions.cpp:26:
159 /home/kaib/OPM/debug/opm-parser/opm/parser/eclipse/EclipseState/EclipseState.hpp:45:11: error: forward declaration of ‘const class Opm::Schedule’
160      class Schedule;
161            ^
162 /home/kaib/OPM/debug/dune-cornerpoint/dune/grid/common/ZoltanGraphFunctions.cpp:345:53: error: invalid use of incomplete type ‘const class Opm::Schedule’
163      int last_time_step = eclipseState->getSchedule()->getTimeMap()->size()-1;
164                                                      ^
165 In file included from /home/kaib/OPM/debug/dune-cornerpoint/dune/grid/common/ZoltanGraphFunctions.hpp:24:0,
166                  from /home/kaib/OPM/debug/dune-cornerpoint/dune/grid/common/ZoltanGraphFunctions.cpp:26:
167 /home/kaib/OPM/debug/opm-parser/opm/parser/eclipse/EclipseState/EclipseState.hpp:45:11: error: forward declaration of ‘const class Opm::Schedule’
168      class Schedule;
169            ^
170 /home/kaib/OPM/debug/dune-cornerpoint/dune/grid/common/ZoltanGraphFunctions.cpp:354:9: error: ‘CompletionSetConstPtr’ is not a member of ‘Opm’
```

**Comment by joakim-hove** [comment]
Hmmm;

I'm sorry there has been so much build problems for you lately @GitPaean. My hunch is that this is due to unsufficent build testing *with MPI*. Tomorrow.

**Comment by jokva** [comment]
@GitPaean What cmake configuration options do you use to compile dune-cornerpoint?

**Comment by GitPaean** [comment]
I think the you need to have parallel version of DUNE and Zoltan. 

```
  1 set(CMAKE_C_COMPILER "mpicc" CACHE STRING "C compiler")
  2 set(CMAKE_CXX_COMPILER "mpicxx" CACHE STRING "C++ compiler")
  3 set(CMAKE_Fortran_COMPILER "mpif90" CACHE STRING "Fortran compiler")
  6 set(PRECOMPILE_HEADERS OFF CACHE BOOL "Precompiled headers")
  7 set(SILENCE_EXTERNAL_WARNINGS ON CACHE BOOL "Silence warnings from external libraries")
  8 set(USE_MPI ON CACHE BOOL "Use MPI")
  9 set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "Build type")
 10 set (CMAKE_CXX_FLAGS
 11         "-pipe -std=c++0x -pedantic -Wall -Wextra -Wformat-nonliteral -Wcast-align -Wpointer-arith -Wmissing-declarations -Wcast-qual -Wshadow -Wwrite-strings -Wchar-subscripts -Wredundant-decls"
 12         CACHE STRING "Flags used by the compiler during all build types")
 13 set (CMAKE_C_FLAGS_DEBUG
 14         "-O2 -g -fno-omit-frame-pointer -march=native -mtune=native"
 15         CACHE STRING "Flags used during Debug builds")
 16 set (CMAKE_CXX_FLAGS_DEBUG
 17         "-O2 -g -fno-omit-frame-pointer -march=native -mtune=native"
 18         CACHE STRING "Flags used during Debug builds")
```

**Comment by jokva** [comment]
@GitPaean Could you test https://github.com/OPM/dune-cornerpoint/pull/202 ? I think it should fix your issues.

Unfortunately I don't have Zoltan readily available, so it's somewhat untested. I'll try to get a working, similar setup, but you can test this far quicker than I can.

**Comment by GitPaean** [comment]
fixed by #202. closing now.

**Comment by jokva** [comment]
Excellent! Thanks for your help & report.
